dependencies/editline/meson.build

```diff
@@ -0,0 +1,33 @@
+# Editline Dependency File
+#============================================================================
+
+
+# Look for editline, a required dependency.
+#--------------------------------------------------
+# NOTE: The the libeditline.pc file was added only in libeditline >= 1.15.2, see 
+# https://github.com/troglobit/editline/commit/0a8f2ef4203c3a4a4726b9dd1336869cd0da8607,
+# but e.g. Ubuntu 16.04 has an older version, so we fall back to searching for
+# editline.h when the pkg-config approach fails.
+
+editline_dep = declare_dependency(
+    dependencies : cpp.find_library('editline'),
+    link_args :  get_option('editline_link_args'))
+
+
+if not (
+    cpp.has_function(
+        'read_history',
+        prefix : '#include <stdio.h>\n#include "editline.h"',
+        args : '-lreadline'))
+    warning('Nix requires libeditline; However, required functions do not work. Maybe ' +\
+    'it is too old? >= 1.14 is required.')
+endif
+
+
+if (cpp.has_header('editline.h'))
+    config_h.set(
+        'HAVE_EDITLINE_H', 1,
+        description : 'Define to 1 if you have the <editline.h> header file.')
+else
+    error('Nix requires editline.h; however the header was not found.')
+endif
```

Comment:
It seems like these checks are in reverse order? Alternatively, if you require Meson >= 0.50, you can use
```meson
cpp.find_library('editline', has_headers : ['editline.h'])
```

Reply:
looks a lot cleaner. Don't think that was added into meson when i wrote this :)

Reply:
just revisited this; its checking for the function `read_history`. which isnt available in older versions of editline. It's not checking for the existence of a header file

Reply:
@dcbaker maybe a cleaner way of doing it would to enable has_headers for has_function() ?

Reply:
The problem here is that you do the following:
1. check for libeditline
2. check for read_line, using editline.h
3. check that editline.h exists

I'd probably write this check as:
```meson
dep_editline = cc.find_library('editline', has_headers : ['editline.'h])
if not cpp.has_function(
    'read_line',
     prefix : '#include <stdio.h>\n#include "editline.h"',
     dependencies : dep_editline)
  error(...)
endif
```

Reply:
i feel it is best to mirror the existing buildsystem as closely as possible, and seperately check for the header and functions; the header check should be before the function check though :)